q2_feature_table/_filter.py

```diff
@@ -13,7 +13,6 @@
 
 def _get_biom_filter_function(ids_to_keep, min_frequency, max_frequency,
                               min_nonzero, max_nonzero):
-    ids_to_keep = set(ids_to_keep)
```

Comment:
By the time we get here, `ids_to_keep` have become a set no matter what, so this is redundant.